Ba6_MiniProjet/main.c

```diff
@@ -53,63 +51,68 @@ int main(void) {
 	serial_start();
 	//starts the USB communication
 	usb_start();
-    //starts the camera
-    dcmi_start();
+	//starts the camera
+	dcmi_start();
 	po8030_start();
 	process_image_start();
 	//start the bodyled thread
 	body_led_thd_start();
 	init_potentiometer();
 
-    //starts timer 12
-    timer12_start();
-    //inits the motors
-    motors_init();
-    //start the ToF distance sensor
-    VL53L0X_start();
+	//starts timer 12
+	timer12_start();
+	//inits the motors
+	motors_init();
+	//start the ToF distance sensor
+	VL53L0X_start();
+	init_pi_regulator();
 	uint16_t distance = 0;
 	//start the spi for the rgb leds
 	spi_comm_start();
 
-
-
 //    //temp tab used to store values in complex_float format
 //    //needed bx doFFT_c
 //    static complex_float temp_tab[FFT_SIZE];
 //    //send_tab is used to save the state of the buffer to send (double buffering)
 //    //to avoid modifications of the buffer while sending it
 //    static float send_tab[FFT_SIZE];
 
-    //starts the microphones processing thread.
-    //it calls the callback given in parameter when samples are ready
-    mic_start(&processAudioData);
+	//starts the microphones processing thread.
+	//it calls the callback given in parameter when samples are ready
+//	mic_start(&processAudioData);
 
 	/* Infinite loop. */
 	while (1) {
 
 		switch (get_punky_state()) {
 		case PUNKY_DEMO:
+			set_rgb_led(LED8, 0, 100, 0);
+
 			//audio processing
 
 			//code barre
 
-
 			//laser
-			distance = VL53L0X_get_dist_mm();
-//			chprintf((BaseSequentialStream *) &SD3, "distance %d    ", distance);
-			if (distance > min_dist_barcode && distance < max_dist_barcode){
+			if (distance < max_dist_barcode) {
+				set_rgb_led(LED8, 0, 100, 0);
 				get_images();
 				set_led(LED5, 1);
-			}else{
-				stop_images();
-				set_led(LED5, 0);
+				deactivate_motors();
+				start_pi_regulator();
```

Comment:
J'imagine que c'est normal puisque tu as dis ne pas avoir touché au codebarre ?
au niveau du if else il y a un bug car actuellement c'est noté soit on est au pi-regulator soit à la lecture de code barre alors qu'ils sont en même temps

Reply:
on peut recheck mardi pour etre sur, mais il me semble que ca doit etre bon